### PR TITLE
Use pkgdown website

### PR DESCRIPTION
**Pull request by schloerke**
This PR does **NOT** improve the documentation.  

The goal of the PR is to transfer content from https://github.com/rstudio/apis-plumber/ to this repo.  This will allow for all developer documentation and _book_ documentation to be hosted within the same repo.

Work still needs to be done in presenting the reference page and other content tweaks. 

PR task list:
- [x] Update NEWS
- [NA] Add tests
- [x] Update documentation with `devtools::document()`

### CODE REVIEW

File: R/plumber-step.R [hunk 1]
```diff
@@ -162,7 +162,13 @@ PlumberEndpoint <- R6Class(
     #' @param req a request object
     #' @return a logical. `TRUE` when endpoint can serve request.
     canServe = function(req){
-      req$REQUEST_METHOD %in% self$verbs && !is.na(stri_match_first_regex(req$PATH_INFO, private$regex$regex)[1,1])
+      req$REQUEST_METHOD %in% self$verbs && self$matchesRoute(req$PATH_INFO)
```
Comment (meztez) [inline R/plumber-step.R:165]: I think this is a spillover from 405, anyway this will fail most test as `matchesRoutes` is not `matchesPath` below.

**Comment by meztez** [comment]
So excited for the documentation to be using pkgdown, great job @schloerke .

File: R/plumber-step.R (hunk 1 above)
Comment (schloerke) [reply R/plumber-step.R]: Thank you! I was confused there for a min.. haha

File: .github/workflows/pkgdown.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+          remotes::install_dev("pkgdown")
+
+      - name: Install package
+        run: R CMD INSTALL .
+
+      - name: Build Site (PR)
+        if: github.event_name != 'push'
+        shell: Rscript {0}
+        run: |
+          pkgdown::build_site()
```
Comment (cpsievert) [inline .github/workflows/pkgdown.yaml:49]:
```suggestion
          pkgdown::build_site(new_process = FALSE)
```

File: README.md [hunk 3]
```diff
@@ -1,4 +1,4 @@
-# plumber
+# plumber <a href='https://rstudio.github.io/plumber'><img src='man/figures/logo.png' align="right" height="138.5" style="margin:10px;" /></a>
```
Comment (cpsievert) [inline README.md:1]: Maybe consider using svg instead

File: vignettes/files/apis/03-01-dynamic.R [hunk 4]
```diff
@@ -0,0 +1,10 @@
+users <- data.frame(
+  uid=c(12,13),
+  username=c("kim", "john")
+)
+
+#' Lookup a user
+#' @get /users/<id>
+function(id){
+  subset(users, uid==id)
```
Comment (cpsievert) [inline vignettes/files/apis/03-01-dynamic.R:9]:
```suggestion
  subset(users, uid %in% id)
```

File: vignettes/toc.css [hunk 5]
```diff
@@ -0,0 +1,127 @@
+#TOC ul,
```
Comment (cpsievert) [inline vignettes/toc.css:1]: Can delete

File: vignettes/style.css [hunk 6]
```diff
@@ -0,0 +1,14 @@
+p.caption {
+  color: #777;
+  margin-top: 10px;
+}
```
Comment (cpsievert) [inline vignettes/style.css:4]: Can delete

**Comment by cpsievert** [comment]
Should also setup netlify redirects so that a https://www.rplumber.io/docs links redirects to the right location 

See some examples at https://github.com/cpsievert/plotly_book/blob/master/docs/_redirects

**Comment by schloerke** [comment]
Must wait until Jeff changes netlify settings

**Pull request merged**